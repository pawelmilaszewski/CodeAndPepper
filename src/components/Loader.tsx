import React, {FC} from 'react';
import {View, ActivityIndicator, StyleSheet} from 'react-native';

import {Colors} from '../utils/colors';

interface Props {
  loading: boolean;
}

export const Loader: FC<Props> = ({loading}): JSX.Element => {
  return (
    <>
      {loading && (
        <View style={styles.main} testID={'loader'}>
          <ActivityIndicator
            animating={loading}
            size={'large'}
            color={Colors.BLACK}
          />
        </View>
      )}
    </>
  );
};

const styles = StyleSheet.create({
  main: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: `${Colors.BLACK}75`,
    position: 'absolute',
    top: 0,
    bottom: 0,
    left: 0,
    right: 0,
    zIndex: 10,
  },
});
