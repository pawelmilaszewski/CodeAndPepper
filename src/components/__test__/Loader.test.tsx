import React from 'react';
import Loader from '../Loader';
import {render} from '@testing-library/react-native';

test('render custom header button', () => {
  const tree = render(<Loader loading={true} />);

  const {toJSON, getByTestId} = tree;

  const result = getByTestId('loader');
  expect(result).toBeTruthy();

  expect(toJSON()).toMatchSnapshot();
});
