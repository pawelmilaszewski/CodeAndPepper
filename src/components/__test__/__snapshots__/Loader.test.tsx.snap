// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render custom header button 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#00000075",
      "bottom": 0,
      "flex": 1,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 10,
    }
  }
  testID="loader"
>
  <ActivityIndicator
    animating={true}
    color="#000000"
    size="large"
  />
</View>
`;
